### ISSUE TITLE
Shell module calling fish instead of /bin/sh

### ISSUE THREAD

**Issue by ElijahLynn** [ansible/ansible#32523]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
Shell module

##### ANSIBLE VERSION
```
ansible 2.4.0 (detached HEAD 91a72ce7da) last updated 2017/11/02 16:14:31 (GMT -400)
  config file = home/ansible/ansicloud/playbooks/cms/sandbox/ansible.cfg
  configured module search path = [u'/home/elijah/bin/ansible/library']
  ansible python module location = /home/elijah/bin/ansible/lib/ansible
  executable location = /home/elijah/bin/ansible/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### OS / ENVIRONMENT
Ubuntu 14.04

##### SUMMARY
When running a task using the Shell module with `&&` I get an error saying fish can't use `&&`, which is true, it can't. But why is Shell module trying to use fish? http://docs.ansible.com/ansible/latest/shell_module.html says it should be using /bin/sh, which does exist on my system. 

##### EXPECTED RESULTS
The command to run in /bin/sh

##### ACTUAL RESULTS
The command ran in my default shell, fish

**Comment by ElijahLynn** [comment]
Update: Using `executable: /bin/sh` does work but I wasn't expecting to have to specify that in all our tasks. Should it be calling `/bin/sh` locally as well as remotely?

**Comment by bcoca** [comment]
I think this i a dupe we already fixed, check against 2.4.2 (currently in beta)

**Comment by ElijahLynn** [comment]
@bcoca Thanks, 2.4.2 beta 2 does appear to work. Now I have to get our dynamic inventory script figured out so it doesn't throw an aws_region undefined error past 91a72ce7da so I can actually use 2.4.2.
